classification.yaml

```diff
@@ -0,0 +1,791 @@
+DMS-MWBT-REQ-0085:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Run QG generation against any major repo (e.g. `/repo/main`).
+  reuse_script: DMS-MWBT-REQ-0083
+DMS-MWBT-REQ-0025:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Make an empty repo with the default configuration.
+    - Make an empty repo with configuration that overrides a formatter.
+    - Make an empty repo with configuration that changes a StorageClass's Python type.
+    - Put and get the same datasets to all repos.
+DMS-MWBT-REQ-0026:
+  status: hackable
+  priority: 1a
+  script:
+    - Add a butler method that prints/saves the full expanded config (it's just a private attribute right now).
+    - Use it.
+DMS-MWBT-REQ-0027:
+  status: feature-incomplete
+  priority: 1a
+  script:
+    - Run pipeline with repo config override that uses InMemoryDatastore.
+  missing:
+    - Need to resolve issues with InMemoryDatastore, or at least those inside the middleware (we could live with the "use at your own risk" aliasing issue, as that's probably impossible to really fix).
+DMS-MWBT-REQ-0030:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Make an empty repo with a POSIX datastore, do get/put.
+DMS-MWBT-REQ-0031:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Make an empty repo with an S3 datastore, do get/put.
+DMS-MWBT-REQ-0028:
+  status: feature-incomplete
+  priority: 1a
+  script:
+    - Make an empty repo with a VOSpace datastore, do get/put.
+  missing:
+    - A VOSpace datastore implementation.
+  notes:
+    Is this actually distinct from S3?
+DMS-MWBT-REQ-0029:
+  status: feature-incomplete
+  priority: 1a
+  script:
+    - Make an empty repo with a VOSpace datastore, do get/put.
+  reuse_script: DMS-MWBT-REQ-0028
+  missing:
+    - A VOSpace datastore implementation.
+  notes:
+    Is this actually distinct from S3?
+DMS-MWBT-REQ-0032:
+  status: unclear
+  priority: 1a
+  notes:
+    Looking at the motivating use case, this requirement seems to really be about supporting both HDF5 and FITS.  But the spec seems to suggest we need to be able to support "any" scientific data format, which is not really enumerable.
+    It also seems to presume that we would have some classes that can be serialized identically to different formats.  But we don't, and I don't really see us adding any just to satisfy this test.
+DMS-MWBT-REQ-0033:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Use a butler to read a subimage via get parameters against any repo.
+DMS-MWBT-REQ-0034:
+  status: feature-complete
+  priority: 1a
+  script:
+    - Use a butler to read a component (e.g. WCS), against any repo.
+DMS-MWBT-REQ-0035:
+  status: unclear
+  priority: 1a
+  notes:
+    This is not a middleware responsibility; it's a feature of whatever in-memory dataset one is trying to operate on (e.g. afw classes).
```

Comment:
This is probably a contentious one; I think DM needs to deliver the high-level functionality, but I'm not sure _under_ butler or in middleware is a good place for it.  Middleware already provide low-level functionality that could be used by some service to provide the functionality on top of butler.

Reply:
Doesn't this one require virtual datasets and that seems clearly a middleware functionality rather than a user bolt on. Are you suggesting the alternative that someone does a calexp.wcs get then does a raw get and then replaces the WCS of the raw with the calexp. That makes things more complicated than they would be if we had virtual datasets.  I was envisaging the ability for virtual datasets to be created in registry and then you could point your pipeline at the collection of raws that had the fixed WCS (that sounds easier than having a new dataset type since all the pipelines expect `raw`).

Reply:
Is this really a 1a, i.e needed to enter commisioning? I'd at least suggest to change it to a P1b or P2